File: GVO Code v3/quad3to1.pas
unit quad3to1;

interface
uses utils, sysutils,windows;
procedure GoQuad3to1(Xcount,Ycount:real);

implementation
uses Main, IO, OMS68SERMC ;

procedure GoQuad3to1(Xcount,Ycount:real);
Var
	CmdStr,CmdStr1, Cmdstr2 : string ;
Begin {3 to 1}
	MovingRa := True;
	MovingDec := True;

	RaPos := xCount ;         //is positive

	Str(Rapos:14:0,CmdStr);
	CmdStr := TrimLeft(CmdStr);
	Str(TrkRate,CmdStr2);
	CmdStr1 := 'AX ST;';              { Send RA STOP}
	SendString(CommRecord,pchar(CmdStr));
//
	CmdStr := 'AX  VL'+xvlslew+' MA'+ CmdStr + ' GD ID; ';      {Send RA SLEW}
	SendString(CommRecord,pchar(CmdStr));
{}
	DecPos := abs(Ycount);
	Str(Decpos:14:0,CmdStr);
	cmdStr := TrimLeft(cmdStr);
	if ( DecTarget < c_lat) then
	begin
		CmdStr := 'AY VL'+yvlslew+' MA-'+ CmdStr + ' GD ID;';
		SendString(CommRecord,pchar(CmdStr));                 {Send DEC SLEW}
	end
	else
	begin
		CmdStr := 'AY VL'+yvlslew+' MA'+ CmdStr + ' GD ID;';
		SendString(CommRecord,pchar(CmdStr));                 {Send DEC SLEW}
	end;
{Scope is SLEWING}
End;  {3 to 1}

end.

File: GVO Code C++ v1/quad3to1.cpp
#include "quad3to1.h"
#include "GlobalValues.h"
#include "CommUtils.h"

#include <cmath>
#include <iomanip>
#include <sstream>
#include <iostream>
#include <thread>
#include <chrono>

bool GoQuad3to1(double Xcount, double Ycount, double decTarget) {
	movingRA = true;
	movingDEC = true;

	RaPos = Xcount;

	std::ostringstream raStream;
	raStream << std::fixed << std::setprecision(0) << RaPos;
	std::string raSteps = raStream.str();

	std::string raCmd = "AX  VL" + xvlslew + " MA" + raSteps + " GD ID;";

	decPos = std::abs(Ycount);

	std::ostringstream decStream;
	decStream << std::fixed << std::setprecision(0) << decPos;
	std::string decSteps = decStream.str();

	std::string decCmd;
	if (decTarget < C_Lat) {
		decCmd = "AY  VL" + yvlslew + " MA-" + decSteps + " GD ID;";
	} else {
		decCmd = "AY  VL" + yvlslew + " MA" + decSteps + " GD ID;";
	}

	std::cout << "GoQuad3to1:\n";
	std::cout << "  Xcount=" << Xcount << " Ycount=" << Ycount << "\n";
	std::cout << "  decTarget=" << decTarget << " C_Lat=" << C_Lat << "\n";
	std::cout << "  RA cmd: [" << raCmd << "]\n";
	std::cout << "  DEC cmd: [" << decCmd << "]\n";

	std::cout << "[GoQuad3to1] before AX send\n";
	bool raOk = SendCommand(raCmd);
	std::cout << "[GoQuad3to1] after AX send, raOk=" << raOk << "\n";

	if (!raOk) {
		std::cout << "Failed to send RA slew command\n";
		return false;
	}

	std::cout << "[GoQuad3to1] before AY send\n";
	bool decOk = SendCommand(decCmd);
	std::cout << "[GoQuad3to1] after AY send, decOk=" << decOk << "\n";

	if (!decOk) {
		std::cout << "Failed to send DEC slew command\n";
		return false;
	}

	return true;
}